mscore/instrwidget.cpp

```diff
@@ -590,85 +723,93 @@ void InstrumentsWidget::on_addButton_clicked()
-                  partiturList->blockSignals(true);
-                  delete item; // fires selectionChanged too early (item is still in view)
-                  partiturList->blockSignals(false);
-                  emit on_partiturList_itemSelectionChanged(); // fire manually (item gone by now)
+                  if (countTopLevelItems == 1) {
+                        //Q_ASSERT(false); // shouldn't get here as (remove button disabled when one item left)
+                        removeButton->setEnabled(false); // nevertheless, handle gracefully in release builds
+                        emit completeChanged(false);
+                        return;
                   }
-            else {
-                  ((PartListItem*)item)->op = ListItemOp::I_DELETE;
-                  item->setHidden(true);
+                  int idx = partiturList->indexOfTopLevelItem(item);
+                  if (((PartListItem*)item)->op == ListItemOp::ADD) {
+                        partiturList->blockSignals(true);
+                        delete item; // fires selectionChanged too early (item is still in view)
+                        partiturList->blockSignals(false);
+                        emit on_partiturList_itemSelectionChanged(); // fire manually (item gone by now)
                   }
-            // select an item, do not consider hidden ones
-            int plusIdx = 0;
-            QTreeWidgetItem* nextParent = partiturList->topLevelItem(idx + plusIdx);
-            while (nextParent && nextParent->isHidden()) {
-                  plusIdx++;
-                  nextParent = partiturList->topLevelItem(idx + plusIdx);
+                  else {
+                        ((PartListItem*)item)->op = ListItemOp::I_DELETE;
+                        item->setHidden(true);
                   }
-            if(!nextParent) { // couldn't find one after, check before
-                  plusIdx = 1;
-                  nextParent = partiturList->topLevelItem(idx - plusIdx);
+                  // select an item, do not consider hidden ones @TODO: remove this and let QTreeWidget handle it.
+                  int plusIdx = 0;
+                  QTreeWidgetItem* nextParent = partiturList->topLevelItem(idx + plusIdx);
                   while (nextParent && nextParent->isHidden()) {
                         plusIdx++;
+                        nextParent = partiturList->topLevelItem(idx + plusIdx);
+                  }
+                  if(!nextParent) { // couldn't find one after, check before
+                        plusIdx = 1;
                         nextParent = partiturList->topLevelItem(idx - plusIdx);
+                        while (nextParent && nextParent->isHidden()) {
+                              plusIdx++;
+                              nextParent = partiturList->topLevelItem(idx - plusIdx);
                         }
                   }
-            partiturList->setCurrentItem(nextParent);
+                  }
```

Comment:
Two braces shouldn't have to appear above each other like this :)

... it's indentation again :)